lib/mongoose-core.module.ts

```diff
@@ -191,13 +195,16 @@ export class MongooseCoreModule implements OnApplicationShutdown {
     uri: string,
     mongooseOptions: ConnectOptions,
     lazyConnection?: boolean,
+    onConnectionCreate?: MongooseModuleOptions['onConnectionCreate'],
```

Comment:
Can we declare a one options argument (`factoryOptions`) which has two optional attributes: `lazyConnection` and `onConnectionCreate`?

Reply:
@kamilmysliwiec I've made the updates and also ensured that I add a `@deprecated` comment for the older `lazyConnection` option.

Please let me know if you'd like for me to remove the option completely and only have `factoryOptions`. I feel we can remove it in future versions. Hopefully v11 or v12.